fastcrypto/src/groups/mod.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+/// Impl operator overload for the elements of a given group for all combinations of owned and borrowed
+/// values.
+/// * `scalar` -  Type of scalars. Should be equal to group::Scalar.
+/// * `element` - Type of elements. Should be equal to group::Element.
+/// * `group` - Type of group.
+#[macro_export]
+macro_rules! impl_group {
+    ($scalar:ident, $element:ident, $group:ident) => {
```

Comment:
I haven't been able to write a macro that gets the element and scalar types from the group, eg. $group::Element and $group::Scalar, so they have to be given as arguments for now.

Reply:
We should chat, this is a general place where proc_macros shine.

Reply:
Sure!

Reply:
I've committed a proc_macro to overload the ops now. Let me know what you think!